api/types/versions/context.go

```diff
@@ -0,0 +1,33 @@
+package versions
+
+import (
+	"context"
+)
+
+// APIVersionKey is the API version used for the request's context.
+//
+// Deprecated: use [WithVersion].
+type APIVersionKey = apiVersionKey
+
+// apiVersionKey is the API version used for the request's context.
+type apiVersionKey struct{}
+
+// WithVersion adds the API version on the context. It panics if ctx is
+// a nil-context.
+func WithVersion(ctx context.Context, version string) context.Context {
+	return context.WithValue(ctx, apiVersionKey{}, version)
+}
```

Comment:
Actually, let me change this one to _not_ set the value if `version` is empty; that makes it symmetrical with `FromContext`, and also keeps the option open to better disambiguate "no version set".